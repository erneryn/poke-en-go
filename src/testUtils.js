import { createStore, applyMiddleware } from 'redux';
import reducers from './store/reducers'

import thunk from 'redux-thunk'

export const storeFactory = (initialstate)=>{
    return createStore(reducers,initialstate,applyMiddleware(thunk))
}

export const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }