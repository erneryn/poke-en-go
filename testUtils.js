import { createStore, applyMiddleware } from 'redux';
import reducers from './src/store/reducers'

import thunk from 'redux-thunk'

export const storeFactory = (initialstate)=>{
    return createStore(reducers,initialstate,applyMiddleware(thunk))
}