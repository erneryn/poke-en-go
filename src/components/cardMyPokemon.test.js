import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import { findByTestAttr  } from '../testUtils';
import CardMoyPokemon from './cardMyPokemon'
const defaultProps = { data: [], reload : ()=> true };

const setup = (props={}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<CardMoyPokemon {...setupProps} />)
}

test('renders without error', () => {
  const wrapper = setup();
  const component = findByTestAttr(wrapper, 'component-cardmypokemon');
  expect(component.length).toBe(1);
});

test('card my pokemon to match snapshot',()=>{
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot()
})
