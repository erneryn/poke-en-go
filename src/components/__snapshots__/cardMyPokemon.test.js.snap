// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card my pokemon to match snapshot 1`] = `
<div
  className="item"
  data-test="component-cardmypokemon"
>
  <div
    className="card"
  >
    <div
      className="head"
    >
      <div
        className="name"
      />
      <div
        className="nickname"
      />
    </div>
    <div>
      <img
        alt=""
      />
    </div>
    <div
      className="footer"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
